OpenSim/Common/Test/testTableSource.cpp

```diff
@@ -0,0 +1,216 @@
+    void connect() {
+        Super::connect(*this);
+    }
+    void buildUpSystem(SimTK::MultibodySystem& system) { 
+        connect();
+        addToSystem(system);
+    }
+
+    const SimTK::SimbodyMatterSubsystem& getMatterSubsystem() const { 
+        return *matter; 
+    }
+
+    SimTK::SimbodyMatterSubsystem& updMatterSubsystem() const { 
+        return *matter; 
+    }
+
+    const SimTK::GeneralForceSubsystem& getForceSubsystem() const { 
+        return *forces; 
+    }
+
+    SimTK::GeneralForceSubsystem& updForceSubsystem() const { 
+        return *forces; 
+    }
+
+protected:
+    // Component interface implementation
+    void extendAddToSystem(SimTK::MultibodySystem& system) const override {
+        if (system.hasMatterSubsystem()){
+            matter = system.updMatterSubsystem();
+        }
+        else{
+            // const Sub& subc = getMemberSubcomponent<Sub>(intSubix);
+
+            SimTK::SimbodyMatterSubsystem* old_matter = matter.release();
+            delete old_matter;
+            matter = new SimTK::SimbodyMatterSubsystem(system);
+
+            SimTK::GeneralForceSubsystem* old_forces = forces.release();
+            delete old_forces;
+            forces = new SimTK::GeneralForceSubsystem(system);
+
+            SimTK::Force::UniformGravity gravity(*forces, *matter, 
+                                                 SimTK::Vec3(0, -9.816, 0));
+            fix = gravity.getForceIndex();
+
+            system.updMatterSubsystem().setShowDefaultGeometry(true);
+        }
+    }
+
+private:
+    // Keep track of pointers to the underlying computational subsystems. 
+    mutable SimTK::ReferencePtr<SimTK::SimbodyMatterSubsystem> matter;
+    mutable SimTK::ReferencePtr<SimTK::GeneralForceSubsystem> forces;
+
+    // keep track of the force added by the component
+    mutable SimTK::ForceIndex fix;
+
+    MemberSubcomponentIndex intSubix{constructSubcomponent<Sub>("internalSub")};
+}; // class TheWorld
+
+
+template<typename RowVec>
+void assertEqual(const RowVec& a, const RowVec& b) {
+    assert(a.nrow() == b.nrow());
+    assert(a.ncol() == b.ncol());
+    for(int i = 0; i < a.ncol(); ++i)
+        ASSERT_EQUAL(a[i], b[i], 1e-10);
+}
+
+int main() {
+    using namespace OpenSim;
+    using namespace SimTK;
+
+    TimeSeriesTable table{};
+    table.setColumnLabels({"0", "1", "2", "3"});
+    SimTK::RowVector_<double> row{4, double{0}};
+    for(unsigned i = 0; i < 4; ++i)
+        table.appendRow(0.00 + 0.25 * i, row + i);
+
+    std::cout << "Contents of the table :" << std::endl;
+    std::cout << table << std::endl;
+
+    auto tableSource = new TableSource{table};
+
+    auto tableReporter = new TableReporter_<double, double>{};
+
+    // Define the Simbody system
+    MultibodySystem system;
+
+    TheWorld theWorld;
+    theWorld.setName("World");
+
+    theWorld.add(tableSource);
+    theWorld.add(tableReporter);
+
+    tableReporter->updInput("inputs").connect(tableSource->getOutput("column"));
```

Comment:
There seems to be a mismatch here where `getOutput("column")` returns a single column/channel and `updInput("inputs")` seems to be a list of inputs, so it isn't clear that all columns are being wired up to separate channels of the inputs. Otherwise it seems like a single "column" output is satisfying one input.